core/src/main/java/hudson/PluginManager.java

```diff
@@ -1829,6 +1840,34 @@ public MetadataCache createCache() {
         return new MetadataCache();
     }
 
+    /**
+     * Disable a list of plugins using a strategy for their dependants plugins.
+     * @param strategy the strategy regarding how the dependant plugins are processed
+     * @param plugins the list of plugins
+     * @return the list of results for every plugin and their dependant plugins.
+     * @throws IOException see {@link PluginWrapper#disable()}
+     */
+    public List<PluginWrapper.PluginDisableResult> disablePlugins(PluginWrapper.PluginDisableStrategy strategy, List<String> plugins) throws IOException {
+        // Where we store the results of each plugin disablement
+        List<PluginWrapper.PluginDisableResult> results = new ArrayList<>(plugins.size());
+
+        // Disable all plugins passed
+        for (String pluginName : plugins) {
+            PluginWrapper plugin = this.getPlugin(pluginName);
+
+            if (plugin == null) {
+                PluginWrapper.PluginDisableResult result = new PluginWrapper.PluginDisableResult(pluginName);
+                result.setMessage(Messages.PluginWrapper_NoSuchPlugin(pluginName));
```

Comment:
It seems weird to create an object and then set some attributes when you already knew those when creating the object. Maybe creating a new constructor or a helper method?

Reply:
But you haven't used the new constructor here?